honeycombio/resource_derived_column.go

```diff
@@ -66,7 +73,13 @@ func resourceDerivedColumnCreate(ctx context.Context, d *schema.ResourceData, me
 	dataset := d.Get("dataset").(string)
 	derivedColumn := readDerivedColumn(d)
 
-	derivedColumn, err := client.DerivedColumns.Create(ctx, dataset, derivedColumn)
+	existing, err := client.DerivedColumns.GetByAlias(ctx, dataset, derivedColumn.Alias)
+	if err == nil {
+		d.SetId(existing.ID)
+		return resourceDerivedColumnUpdate(ctx, d, meta)
+	}
+
+	derivedColumn, err = client.DerivedColumns.Create(ctx, dataset, derivedColumn)
```

Comment:
Same here, we don't need to get and store the ID because it is set by `resourceDerivedColumnRead`.